runtime-extension-model/src/main/java/org/mule/runtime/core/internal/extension/ForEachChainInputTypeResolver.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.extension;
+
+import static java.lang.String.format;
+
+import org.mule.metadata.api.model.ArrayType;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.message.api.MessageMetadataType;
+import org.mule.metadata.message.api.MessageMetadataTypeBuilder;
+import org.mule.sdk.api.metadata.ChainInputMetadataContext;
+import org.mule.sdk.api.metadata.resolving.ChainInputTypeResolver;
+
+/**
+ * {@link ChainInputTypeResolver} for the ForEach and ParallelForEach scopes.
+ *
+ * @since 4.8.0
+ */
+public class ForEachChainInputTypeResolver implements ChainInputTypeResolver {
+
+  private static final String PARAMETER_NAME = "collection";
```

Comment:
overkill

Reply:
Do you mean the constant? but if I don't define it, the same string would be repeated in two places. I'd like to make sure they are always in sync.

Reply:
but you don't reuse it. This string will be kept in memory forever, even at runtime.

Reply:
I do use it, [here](https://github.com/mulesoft/mule/pull/13602/files/788389de81e8038c5ed2d1759b8de6aae5ff4b19#diff-d357c4ff11c93c077669d118a359fb1830ade8166c9234536ea40ce7a02646e2R35) and [here](https://github.com/mulesoft/mule/pull/13602/files/788389de81e8038c5ed2d1759b8de6aae5ff4b19#diff-d357c4ff11c93c077669d118a359fb1830ade8166c9234536ea40ce7a02646e2R52).
I haven't checked the details of how the JVM does this, but I would assume that even if I leave the string literals inlined in the code where they are used, those literals would be in memory as long as the class is in memory, same as the static variable. But it could be that they go to different memory sections.

But I don't think it is worth extending the debate for such a small thing, so I will just change it.

Reply:
We agreed to keep it as is. Marking as resolved.